torch/_tensor.py

```diff
@@ -1093,70 +1093,72 @@ def unique_consecutive(self, return_inverse=False, return_counts=False, dim=None
             self, return_inverse=return_inverse, return_counts=return_counts, dim=dim
         )
 
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __rsub__(self, other):
-        return _C._VariableFunctions.rsub(self, other)
-
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __rdiv__(self, other):
-        return self.reciprocal() * other
-
-    __rtruediv__ = __rdiv__
-    __itruediv__ = _C.TensorBase.__idiv__
-
-    __pow__ = cast(
-        Callable[
-            ["torch._C.TensorBase", Union["Tensor", int, float, bool, complex]],
-            "Tensor",
-        ],
-        _handle_torch_function_and_wrap_type_error_to_not_implemented(
-            _C.TensorBase.pow
-        ),
-    )
-    __ipow__ = _handle_torch_function_and_wrap_type_error_to_not_implemented(
-        _C.TensorBase.pow_
-    )
-
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __rmod__(self, other):
-        return torch.remainder(other, self)
-
-    def __format__(self, format_spec):
+    def __format__(self, format_spec: str) -> str:
         if has_torch_function_unary(self):
             return handle_torch_function(Tensor.__format__, (self,), self, format_spec)
         if self.dim() == 0 and not self.is_meta and type(self) is Tensor:
             return self.item().__format__(format_spec)
         return object.__format__(self, format_spec)
 
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __rpow__(self, other):
-        return torch.pow(other, self)
+    __pos__ = _C.TensorBase.positive
+    __neg__ = _C.TensorBase.neg
+    __abs__ = _C.TensorBase.abs
 
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __floordiv__(self, other):
-        return torch.floor_divide(self, other)
+    # The typehints for these dunder methods are auto-generated as part of
+    # _C.TensorBase's typestubs, so use those.
+    if not TYPE_CHECKING:
 
-    @_handle_torch_function_and_wrap_type_error_to_not_implemented
-    def __rfloordiv__(self, other):
-        return torch.floor_divide(other, self)
+        @_handle_torch_function_and_wrap_type_error_to_not_implemented
```

Comment:
I don't quite understand why some things need to be inside the `TYPE_CHECKING` block and what thigns don't need to be (and I'm not entirely sure how these tests the `torch/_C/__init__.pyi` file is actually generated from the `__init__.pyi.in` file), but this combination seems to make hte tests pass.